core-examples/rust/message-monitoring/src/main.rs

```diff
@@ -0,0 +1,245 @@
+        ParamsOfGetMonitorInfo, ParamsOfSendMessages,
+    },
+    ClientConfig, ClientContext,
+};
+
+#[derive(Parser, Debug)]
+#[clap(version, about, long_about = None)]
+struct Args {
+    #[clap(long, short, action)]
+    loop_nowait: bool,
+    #[clap(long, short, action)]
+    once_await: bool,
+    #[clap(long, short, action)]
+    await_all: bool,
+    #[clap(long, short, default_value_t = 10)]
+    message_count: u32,
+    #[clap(long, short, default_value_t = 3)]
+    wait_seconds: u32,
+}
+
+const LOOP_SLEEP_IN_SECONDS: u64 = 1;
+const QUEUE: &str = "Foo";
+
+#[tokio::main]
+async fn main() {
+    let args = Args::parse();
+    if !args.await_all && !args.loop_nowait && !args.once_await {
+        let mut cmd = Args::command();
+        cmd.print_help().unwrap();
+        std::process::exit(64);
+    }
+    let giver_abi = json!({
+        "ABI version": 2,
+        "header": ["time", "expire"],
+        "functions": [
+            {
+                "name": "sendTransaction",
+                "inputs": [
+                    {
+                        "name": "dest",
+                        "type": "address",
+                    },
+                    {
+                        "name": "value",
+                        "type": "uint128",
+                    },
+                    {
+                        "name": "bounce",
+                        "type": "bool",
+                    },
+                ],
+                "outputs": [],
+            },
+        ],
+    });
+
+    let network = match env::var("TON_NETWORK_ADDRESS") {
+        Ok(val) => val,
+        Err(_) => "http://localhost:4000/graphql".to_owned(), // gateway endpoint
+    };
+    let address = match env::var("TON_GIVER_ADDRESS") {
+        Ok(val) => val,
+        Err(_) => "0:ece57bcc6c530283becbbd8a3b24d3c5987cdddc3c8b7b33be6e4a6312490415".to_owned(),
+    };
+    let wait_until = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_secs() as u32
+        + args.wait_seconds;
+
+    println!("Message awaiting timeout {} sec.", args.wait_seconds);
+
+    let context = Arc::new(
+        ClientContext::new(ClientConfig {
+            network: NetworkConfig {
+                endpoints: Some(vec![network]),
+                ..Default::default()
+            },
+            ..Default::default()
+        })
+        .unwrap(),
+    );
+
+    let mut giver_keys = match env::var("TON_GIVER_SECRET") {
+        Ok(secret) => nacl_sign_keypair_from_secret_key(
+            context.clone(),
+            ParamsOfNaclSignKeyPairFromSecret { secret },
+        )
+        .unwrap(),
+        Err(_) => KeyPair::new(
+            "2ada2e65ab8eeab09490e3521415f45b6e42df9c760a639bcf53957550b25a16".to_owned(),
+            "172af540e43a524763dd53b26a066d472a97c4de37d5498170564510608250c3".to_owned(),
+        ),
+    };
+    giver_keys.secret.truncate(64);
```

Comment:
Lets add a comment  "The secret key is actually the concatenation of secret and public keys (128 symbols hex string) by design of [NaCL](http://nacl.cr.yp.to/sign.html)."